src/datasets/arrow_dataset.py

```diff
@@ -5138,6 +5150,39 @@ def push_to_hub(
                 "Failed to push_to_hub: please specify either max_shard_size or num_shards, but not both."
             )
 
+        # Create repo if it doesn't exist
+        api = HfApi(endpoint=config.HF_ENDPOINT)
+        try:
+            api.repo_info(repo_id, token=token)
+        except RepositoryNotFoundError:
+            repo_url = create_repo(hf_api=api, repo_id=repo_id, token=token, exist_ok=True, repo_type="dataset")
+            repo_id = get_repo_id_from_repo_url(repo_url)
+
+        # Try to find PR branch if branch is supplied
+        if create_pr and branch is not None:
+            if version.parse(huggingface_hub.__version__) < version.parse("0.9.0"):
+                raise ValueError(
+                    "Using `create_pr` requires `huggingface-hub>=0.9.0`. Please use a more recent version of `huggingface-hub`."
+                )
+            else:
+                for discussion in get_repo_discussions(repo_id, repo_type="dataset"):
```

Comment:
You probably need to import `get_repo_discussions` here. At the top of the file it may fail